src/metaxy/cli/metadata.py

```diff
@@ -19,6 +24,252 @@
 )
 
 
+def _count_lazy_rows(lazy_frame: nw.LazyFrame[Any]) -> int:
+    """Return row count for a Narwhals LazyFrame."""
+    return lazy_frame.select(nw.len()).collect().to_polars()["len"].item()
+
+
+def _preview_rows(
+    lazy_frame: nw.LazyFrame[Any],
+    id_columns: Sequence[str] | None,
+    limit: int = 5,
+) -> list[str]:
+    """Return formatted preview rows for verbose output."""
+    # Determine columns to display
+    headers = list(id_columns or ["sample_uid"])
+    available_headers = [
+        col for col in headers if col in lazy_frame.collect_schema().names()
+    ]
+
+    if not available_headers:
+        # Fallback to all columns if specified headers don't exist
+        available_headers = lazy_frame.collect_schema().names()
+
+    if not available_headers:
+        return []
+
+    # Only collect limited rows
+    preview_df = lazy_frame.select(available_headers).head(limit).collect().to_polars()
+
+    return [
```

Comment:
I'd rather have this returned as Polars (or have another function doing it) s, it can be useful to display this info in other places as well, in different format